htdocs/takepos/index.php

```diff
@@ -817,7 +817,7 @@ function Search2(keyCodeForEnter, moreorless) {
 						console.log("There is only 1 answer with barcode matching the search, so we change the thirdparty "+data[0]['rowid']);
 						ChangeThirdparty(data[0]['rowid']);
 					}
-					else if ($('#search').val() == data[0]['barcode'] && 'product' == data[0]['object']) {
+					else if ('product' == data[0]['object']) {
```

Comment:
It seems the old test was wrong.
According to rhe coment, the old test was doing:
- if we made a search ok a string and result is a product with the barcode searched, we know we did a search to find 1 particular product, in goalntobadd it to basket, so we add it automatically.
- but if we made a search, but not on a barcode value, for example a search on "abc" to see all products that contains abc, we show the result but we do not add the product automatically in basket, even if there is 1 answer ok only (because we want to see answers of search and not necessarly select it, it may depends on result)

And this is the behaviour that we want.
If you change this test, this behaviour will be broken, and if you try to search a product just to see what can be answer on the search string, product will be also added with no chance to check result before.

Am i right ?

Reply:
You're right and I understand your point of view.
However, in the case of you have a bar-code rule and some products with the reference of the product in the 8 first digits, the full bar-code is not equal to the product reference and the 5 others digits are here to determine the quantity and the product must be selected automatically with this quantity.
I believed that if there was only one product found, it will be automatically inserted but I see it not necessary the case (it was the default behavour in Dolibarr v14)
Maybe I have to use another condition for bar-code rule case ?